### PR TITLE
Initiate discovery request from hip to openmrs

### PR DESCRIPTION
**Pull request by davidviktorTW**
https://trello.com/c/56M4IfQI

### CODE REVIEW

File: src/In.ProjectEKA.HipService/OpenMrs/OpenMrsPatientMatchingRepository.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                    request.Patient?.YearOfBirth?.ToString());
+
+            return result
+                    .Select(p => p.ToHipPatient(request.Patient?.Name))
+                    .ToList()
+                    .AsQueryable();
+        }
+    }
+
+    public static class PatientExtensions
```
Comment (barney-spencer) [inline src/In.ProjectEKA.HipService/OpenMrs/OpenMrsPatientMatchingRepository.cs:38]: I think we should move PatientExtensions out to its own file and folder.

Comment (davidviktorTW) [reply src/In.ProjectEKA.HipService/OpenMrs/OpenMrsPatientMatchingRepository.cs]: Done

File: src/In.ProjectEKA.HipLibrary/Patient/Model/Gender.cs [hunk 2]
```diff
@@ -4,6 +4,7 @@ public enum Gender
     {
         M,
         F,
-        O
+        O,
+        U
```
Comment (praveenadayanand) [inline src/In.ProjectEKA.HipLibrary/Patient/Model/Gender.cs:8]: This does not match with the bridge contract. As per the contract Gender can only be M/F/O

File: test/In.ProjectEKA.HipServiceTest/Discovery/PatientControllerTest.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            var requestContent = new DiscoveryRequestPayloadBuilder()
+                .WithMissingParameters(missingRequestParameters)
+                .Build();
+
+            var response =
+                await _client.PostAsync(
+                    "v1/care-contexts/discover",
+                    requestContent);
+
+            Assert.Equal(expectedStatusCode, response.StatusCode);
```
Comment (morgan-tw) [inline test/In.ProjectEKA.HipServiceTest/Discovery/PatientControllerTest.cs:37]: If we want to stay consistent we can use FluentAssertion :
- expectedStatusCode.Should().Be(response.StatusCode); 

Could be done later, not blocking

File: test/In.ProjectEKA.HipServiceTest/Discovery/FilterTest.cs [hunk 4]
```diff
@@ -218,7 +218,7 @@ private Patient BuildLinda()
             return BuildPatient(Linda.Name, Linda.Sex, Linda.PhoneNumber, Linda.YearOfBirth);
         }
 
-        private Patient BuildPatient(string name, Gender gender, string phoneNumber, ushort yearOfBirth)
+        private Patient BuildPatient(string name, Gender? gender, string phoneNumber, ushort? yearOfBirth)
```
Comment (morgan-tw) [inline test/In.ProjectEKA.HipServiceTest/Discovery/FilterTest.cs:221]: At line 211 and 212 we have Faker().Random.String() and RandomString() that come from TestBuilder.
Apparently we may have introduced that together before.

I don't get the point to have a TestBuilder to do that, else we want to get an adapter of test libraries 
Moreover the TestBuilder is duplicated in all domain like DataFlow etc.

May be we can add a card into the Debt Backlog to decide exactly if we use that or remove it :)

File: src/In.ProjectEKA.HipService/Discovery/PatientDiscovery.cs [hunk 5]
```diff
@@ -71,6 +74,7 @@ public class PatientDiscovery
                 DiscoveryUseCase.DiscoverPatient(Filter.Do(patients, request).AsQueryable());
             if (patientEnquiryRepresentation == null)
             {
+                Log.Information($"No mathcing unique patient found for transaction {request.TransactionId}.", error);
```
[the remark this answers is not in the document]
Comment (davidviktorTW) [reply src/In.ProjectEKA.HipService/Discovery/PatientDiscovery.cs:77]: Fixed

File: src/In.ProjectEKA.HipLibrary/Patient/Model/Gender.cs (hunk 2 above)
Comment (davidviktorTW) [reply src/In.ProjectEKA.HipLibrary/Patient/Model/Gender.cs:8]: Removed the U value

File: test/In.ProjectEKA.HipServiceTest/Discovery/PatientControllerTest.cs (hunk 3 above)
Comment (davidviktorTW) [reply test/In.ProjectEKA.HipServiceTest/Discovery/PatientControllerTest.cs:37]: Changed to use FluentAssertions

File: test/In.ProjectEKA.HipServiceTest/Discovery/FilterTest.cs (hunk 4 above)
Comment (davidviktorTW) [reply test/In.ProjectEKA.HipServiceTest/Discovery/FilterTest.cs:221]: I think this is a good point and definitely let's talk about it.
However, it is unrelated to this PR and also a mid-size change, so let's keep it separate.

**Pull request merged**